cmd/configure.go

```diff
@@ -31,19 +31,28 @@ places.
 You can also override certain default settings to suit your preferences.
 `,
 	RunE: func(cmd *cobra.Command, args []string) error {
-		configuration := config.NewConfig()
-
-		viperConfig.AddConfigPath(configuration.Dir)
-		viperConfig.SetConfigName("user")
-		viperConfig.SetConfigType("json")
-		// Ignore error. If the file doesn't exist, that is fine.
-		_ = viperConfig.ReadInConfig()
-		configuration.UserViperConfig = viperConfig
-
+		configuration := LoadUserConfig()
 		return runConfigure(configuration, cmd.Flags())
 	},
 }
 
+// LoadUserConfig wraps the logic for loading the user config
+func LoadUserConfig() config.Config {
```

Comment:
Comparing the logic from the configure and download commands it looks like the download command is now modifying the global viperConfig variable when it doesn't need to.

 Consolidating the loading of viper into a shared function makes sense but it seems like it has to happen outside of the configure command and be a bit more explicit on its intent. 

Looking at all the commands I see the general flow for the initializing the user config is as follows:
- create NewConfig
- Initialize instance of Viper
- Read viper config
- validate user config (error if invalid)
- done

I think the shared function should do just that but either be renamed to indicate it is doing that or be rewritten just to handle the viper loading bits. For example I can see this function moved into the config package with a new function signature

```go
func LoadUserConfigFile() (config.Config, error) {
	cfg := config.NewConfig()
	v := viper.New()
	v.AddConfigPath(cfg.Dir)
	v.SetConfigName("user")
	v.SetConfigType("json")
	// Ignore error. If the file doesn't exist, that is fine.
	_ = v.ReadInConfig()
	cfg.UserViperConfig = v

	if err := validateUserConfig(cfg.UserViperConfig); err != nil {
		return nil, err
	}
	return cfg, nil
}
```
Or Just breakout the viper config parts
```go
func LoadViperConfig(dir, name string) (*viper.Viper, error) {
        v := viper.New()
	v.AddConfigPath(dir)
	v.SetConfigName("user")
	v.SetConfigType("json")
	//Maybe capture and return the error 
	_ = v.ReadInConfig()

	return v, nil
}

// Which would change its usage to something like 
cfg := NewConfig()
cfg.UserViperConfig, err := LoadViperConfig(cfg.Dir, "user")
if err != nil {
  /// handle
}
if err := validateUserConfig(cfg.UserViperConfig); err != nil {
   //handle error as needed for the command
}

```

Reply:
I took your first suggestion, expanding it a bit.